### ISSUE TITLE
`promesa.exec.csp/go` doesn't have channel behavior

### ISSUE THREAD

**Issue by alexandergunnarson** [funcool/promesa#125]
I came across `promesa.exec.csp` and am now using it in a project. Really exciting API and use of JVM virtual threads and all the benefits over `clojure.core.async` that that brings!

In `clojure.core.async`, `go` blocks themselves return a chan. This enables e.g. `(alts!! [(go ...) (timeout ...)])`. However, the same is not the case in Promesa: `(csp/alts! [(csp/go ...) (csp/timeout-chan ...)])` throws an exception because `csp/go` doesn't return a channel, but rather a `deferred`. Should every `csp/go` block return a channel (this would preserve `core.async` behavior, but might be a bit heavy, creating chans unnecessarily), or should there be an explicit `csp/go-chan` (analogously to `csp/timeout-chan`) like so?

```clojure
(require '[promesa.exec.csp :as csp])

(defmacro go-chan
  [& body]
  `(let [c# (csp/chan)]
     (csp/go
       (try
         (when-some [v# (do ~@body)]
           (csp/offer! c# v#))
         (finally
           (csp/close! c#))))
     c#))
```

**Comment by niwinz** [comment]
Hmm, the main reason the `csp/go` returns deferred and not a channel is because the go block can fail, and channel is a bad abstraction for handling return value of single element that can fail.

Additionally, I have never seen the go block return value as part of the `alts!` heheh

Yep, I guess for this use case, the `go-chan` is better than  make the go work in the same way as core.async. The promesa approach already differs from core.async, so I guess this is the way.

Thank you very much for the feedback. 
It should be reiterated that this is experimental and I still reserve the right to make changes that slightly break compatibility. I'm also integrating it on my own project and testing the usability of the library :D

**Comment by niwinz** [comment]
Released `10.0.544` with go-chan and your other fix.

:heart:

**Comment by alexandergunnarson** [comment]
Oh awesome — thanks @niwinz! :D Appreciate your comments above!
